src/uct/tcp/tcp_ep.c

```diff
@@ -1522,6 +1543,16 @@ ucs_status_t uct_tcp_ep_flush(uct_ep_h tl_ep, unsigned flags,
     uct_tcp_ep_t *ep = ucs_derived_of(tl_ep, uct_tcp_ep_t);
     uct_tcp_ep_put_completion_t *put_comp;
 
+    if (ucs_unlikely(flags & UCT_FLUSH_FLAG_CANCEL)) {
+        /* TCP is able to cancel only pending operations, posted TX operations
+         * couldn't be canceled, since some data was already sent to the peer
```

Comment:
what if some `put` is on the flight?

Reply:
we couldn't stop them, don't have rollback mechanism for PUT